packages/hadron-document/src/element.ts

```diff
@@ -775,6 +818,17 @@ export class Element extends EventEmitter {
     return isPlainObject(value) || isArray(value);
   }
 
+  isExpanded(includeChildren = false): boolean {
+    if (includeChildren && this.elements) {
+      return (
+        this.expanded &&
+        this.elements.every((element) => element.isExpanded(includeChildren))
+      );
+    } else {
+      return this.expanded;
+    }
```

Comment:
Can parent be not expanded, but children expanded? That's probably not a possible scenario, right? Can we short circuit here in this case?

Reply:
There is a case where this could apply, first we expand all which will expand all the elements and their child elements and later we collapse one of the child element which has sub elements. Essentially we're collapsing just the child element and not its children. Does it make sense?

Reply:
Why would we ever want that to happen is my question, current behavior seems to be to always collapse children when their parent collapses